packages/babel-generator/src/index.js

```diff
@@ -56,11 +55,16 @@ function normalizeOptions(code, opts): Format {
     jsescOption: {
       quotes: "double",
       wrap: true,
+      minimal: true,
```

Comment:
Note that sometimes user may prefer to preserve the escaped form, especially for the space characters. i.e.
```
"\u{00a0}" // NBSP 
"\u{2005}" // 1/4 em
```
and
https://github.com/angular/angular/blob/9e70bcb34f91d439f5203dc22a44f323d02c4648/packages/compiler/src/ml_parser/html_whitespaces.ts#L19

I suggest we plug in the `minimal` option only when there is no escape sequence inside.

Reply:
I think that it would be better to implement this in `jsesc` (https://github.com/babel/babel/pull/11384#issuecomment-610014446), since it would also benefit non-Babel users.

Reply:
We're only able to detect what the user actually typed via the `raw` value, which is handled before printing the cooked value. If we only have the cooked value, we can only guess what the user intended.

Reply:
For the whitespaces, I think we can add them to the whitelist of the [`minimal` option](https://github.com/mathiasbynens/jsesc#minimal).

Since it changes the default print behaviour, I consider this as a breaking change.

Reply:
The `raw` value already captures this appropriately, though. We only run the cooked value through `jsesc` when there's not a raw value. At that point, we have no idea what the user actually wrote in the source file.

Reply:
Indeed. I do think we want to always escape things like NBSP though.

Maybe we should special-case all non-ASCII entries in `require('unicode-13.0.0/General_Category/Space_Separator/symbols.js');` in jsesc’s `minimal` option.

Reply:
```
[ '0x20',
  '0xa0',
  '0x1680',
  '0x2000',
  '0x2001',
  '0x2002',
  '0x2003',
  '0x2004',
  '0x2005',
  '0x2006',
  '0x2007',
  '0x2008',
  '0x2009',
  '0x200a',
  '0x202f',
  '0x205f',
  '0x3000' ]
```

I think we should continue to print regular space. I could understand escaping the rest, though. @mathiasbynens, are you open to doing that in jsesc?

Reply:
Agreed we wouldn't want to escape any ASCII spaces. We should take non-ASCII entries only.

Definitely open to a patch that does this in jsesc's `minimal` option. We could use [`src/data.js`](https://github.com/mathiasbynens/jsesc/blob/master/src/data.js) to export a list of special-cased symbols, based on reading `require('unicode-13.0.0/General_Category/Space_Separator/symbols.js');` (or `/code-points.js`) and filtering out all ASCII entries.

Reply:
Updated to v3.0.1, which now escapes non-ASCII whitespace.